lib/google_contacts_api/api.rb

```diff
@@ -29,26 +29,101 @@ def get(link, params = {}, headers = {})
         raise UnauthorizedError if defined?(e.response) && self.class.parse_response_code(e.response) == 401
         raise e
       end
-      
+
       # OAuth 1.0 uses Net::HTTP internally
       raise UnauthorizedError if result.is_a?(Net::HTTPUnauthorized)
       result
     end
 
+    # these two methods will replace my previous two, but they aren't wired up to
+    # the rest of the code yet
     # Post request to specified link, with query params
-    # Not tried yet, might be issues with params
-    def post(link, params = {}, headers = {})
-      raise NotImplementedError
-      params["alt"] = "json"
-      @oauth.post("#{BASE_URL}#{link}?#{params.to_query}", headers)
+    def post_v2(link, params = {}, headers = {})
+      merged_params = params_with_defaults(params)
+      uri = "#{BASE_URL}#{link}?#{merged_params.to_query}"
+      begin
+        response = @oauth.post(uri, headers)
+      rescue => e
+        raise UnauthorizedError if defined?(e.response) && self.class.parse_response_code(e.response) == 401
+        raise e
+      end
     end
 
     # Put request to specified link, with query params
-    # Not tried yet
-    def put(link, params = {}, headers = {})
-      raise NotImplementedError
-      params["alt"] = "json"
-      @oauth.put("#{BASE_URL}#{link}?#{params.to_query}", headers)
+    def put_v2(link, params = {}, headers = {})
```

Comment:
For the put method in the api class, I'm having trouble deciding how to form the link to send to the oauth2 gem.  It requires an id number on the end of the link that the post uses.  I'm not sure whether to snip that number out from the id link, or send the entire 'edit' or 'id' link through to the post method.  If I sent the whole link through, it would contradict the style of how the other methods form that link from a constant in the class and from the parameters that are passed in.  I'm also not sure for just passing the id number in, as I'm not sure if trying to match something like "http://www.google.com/m8/feeds/contacts/example%40gmail.com/base/1" with /\/base\/ / to find the number on the end is good either since the user has control over what their email is.
The id link can't be used as a url because it doesn't have 'https' at the start.  I could use the 'edit link' in the put method, however system groups also don't have them.